.github/workflows/main.yml

```diff
@@ -0,0 +1,31 @@
+name: gh-pages
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v1
+      with:
+        fetch-depth: 1
+
+    - name: Build with Node.js
+      uses: actions/setup-node@v1
+      with:
+        node-version: '12.x'
+    - run: npm install
+    - run: npm run build
+
+    - name: GitHub Pages action
+      uses: peaceiris/actions-gh-pages@v2.5.1
+      env:
+        ACTIONS_DEPLOY_KEY: ${{ secrets.ACTIONS_DEPLOY_KEY }}
```

Comment:
What is the deploy key for? Does it just do the git checkin/push?

Every action gets a new GitHub token minted that can use to do the checkin. See https://github.com/mikeal/publish-to-github-action/blob/master/entrypoint.sh

Reply:
just blindly following instructions on this one https://github.com/peaceiris/actions-gh-pages#1-add-ssh-deploy-key
will try it on my account

Reply:
Oh I see what this is, https://github.com/marketplace/actions/github-pages-action#%EF%B8%8F-github_token

Funny enough, I was chatting with someone from GitHub about this today. I think you should just be able to use their GITHUB_TOKEN feature instead. The limitation they are running into is that pushes to master **from other GitHub Actions** won’t trigger push actions, which isn’t something we’re worried about.

I’ll comment in the other thread about the fix for their issue, but I think we can probably just switch to GITHUB_TOKEN.

Reply:
ah, there's an ongoing issue with `GITHUB_TOKEN`: https://github.com/peaceiris/actions-gh-pages/issues/9

the problem is with an action triggering an action, using `GITHUB_TOKEN` won't trigger a gh-pages build on their end, but a custom deploy key will. It works when your repo is private apparently but not public.

https://github.community/t5/GitHub-Actions/Github-action-not-triggering-gh-pages-upon-push/td-p/26869

Perhaps something they'll address, but for now, it seems this is necessary.

Reply:
i think i actually interpreted the issue wrong. github’s own internal infra that triggers the ghpages build won’t fire. that is *super* annoying ;)

Reply:
ok, i setup this deploy key and secret in the repo, it should be ready to go.